app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -312,20 +315,26 @@ public function getCartDataJs($multipage = true)
             $success = Mage::getStoreConfig('payment/boltpay/success');
             $close = Mage::getStoreConfig('payment/boltpay/close');
 
-
             //////////////////////////////////////////////////////
             // Generate and return BoltCheckout javascript.
             //////////////////////////////////////////////////////
+            $immutableQuoteId = ($immutableQuote) ? $immutableQuote->getId() : '';
```

Comment:
is this necessary? I think if $immutableQuote is null there is no chance to step here